nano/node/transport/udp.cpp

```diff
@@ -215,6 +215,21 @@ std::shared_ptr<nano::transport::channel_udp> nano::transport::udp_channels::fin
 	return result;
 }
 
+void nano::transport::udp_channels::clean_node_id (nano::account const & node_id_a)
+{
+	std::lock_guard<std::mutex> lock (mutex);
+	std::vector<nano::endpoint> endpoints;
+	auto existing (channels.get<node_id_tag> ().equal_range (node_id_a));
+	for (auto & record : boost::make_iterator_range (existing))
+	{
+		endpoints.push_back (record.endpoint ());
+	}
+	for (auto & endpoint : endpoints)
+	{
+		channels.get<endpoint_tag> ().erase (endpoint);
```

Comment:
I believe multiindex has a range erase method that can use .first and .second of equal_range directly.